packages/amplify-cli/src/index.ts

```diff
@@ -283,6 +285,54 @@ export async function run() {
   }
 }
 
+async function notifySecurityEnhancementPushRequired(context: Context) {
+  if (FeatureFlags.getBoolean('graphqltransformer.securityEnhancementNotification')) {
+    const providerPlugin = await import(context.amplify.getProviderPlugins(context)?.awscloudformation);
+    const transformerVersion = providerPlugin ? await providerPlugin.getTransformerVersion(context) : -1;
+
+    const meta = stateManager.getMeta();
+
+    const apiNames = Object.entries(stateManager.getMeta()?.api || {})
+      .filter(([_, apiResource]) => (apiResource as $TSAny).service === 'AppSync')
+      .map(([name]) => name);
+
+    if (transformerVersion === 2 && meta.auth && apiNames.length === 1) {
+      const projectPath = pathManager.findProjectRoot() ?? process.cwd();
+
+      let config = stateManager.getCLIJSON(projectPath, undefined, {
+        throwIfNotExist: false,
+        preserveComments: true,
+      });
+
+      const apiName = apiNames[0];
+      const apiResourceDir = path.join(pathManager.getBackendDirPath(), 'api', apiName);
+      const schemaPath = path.join(apiResourceDir, 'schema.graphql');
+
+      fs.appendFile(schemaPath, ' ');
+
+      config.features.graphqltransformer.securityEnhancementNotification = false;
+      stateManager.setCLIJSON(projectPath, config);
+      await FeatureFlags.reloadValues();
+
+      const message = `This version of Amplify CLI introduces additional security enhancements for your GraphQL API. @auth authorization rules applied on primary keys and indexes are scoped down further. The changes are applied automatically with this deployment. This change won't impact your client code. Continue`;
+      const executePush =
```

Comment:
if this is within the aws provider cloudformation then the message can happen on a push.